packages/client/src/events/call.ts

```diff
@@ -1,123 +1,46 @@
-import { CallState, StreamVideoWriteableStateStore } from '../store';
+import { CallingState, CallState } from '../store';
 import { StreamVideoEvent } from '../coordinator/connection/types';
+import { Call } from '../Call';
 
 /**
- * Event handler that watched the delivery of CallAcceptedEvent
- * Updates the state store and notifies its subscribers that
- * the given user will be joining the call.
+ * Event handler that watched the delivery of `call.accepted`.
+ * Once the event is received, the call is joined.
  */
-export const watchCallAccepted = (store: StreamVideoWriteableStateStore) => {
-  return function onCallAccepted(event: StreamVideoEvent) {
+export const watchCallAccepted = (call: Call) => {
+  return async function onCallAccepted(event: StreamVideoEvent) {
     if (event.type !== 'call.accepted') return;
-    const { call_cid } = event;
-    const acceptedIncomingCall = store.incomingCalls.find(
-      (incomingCall) => incomingCall.cid === call_cid,
-    );
-    if (acceptedIncomingCall) {
-      console.warn('Received CallAcceptedEvent for an incoming call');
-      return;
+    const { state } = call;
+    if (state.callingState === CallingState.RINGING) {
+      await call.join();
```

Comment:
@oliverlaz `Call.join` logs error in this case, but it seems a valid scenario. Shouldn't the condition above check, whether the call is already joined / joining?

Reply:
This will execute only if the call hasn't been already accepted.